lightning/src/ln/channelmanager.rs

```diff
@@ -4358,6 +4358,111 @@ where
 			)
 	}
 
+	fn get_response_for_invoice_request(&self, invoice_request: InvoiceRequest, context: Option<OffersContext>, custom_amount_msats: Option<u64>) -> Result<(OffersMessage, PaymentHash), Bolt12ResponseError> {
+		let secp_ctx = &self.secp_ctx;
+		let expanded_key = &self.inbound_payment_key;
+
+		// Make sure that invoice_request amount and custom amount are not present at the same time.
+		if invoice_request.amount().is_some() && custom_amount_msats.is_some() {
+			return Err(Bolt12ResponseError::UnexpectedAmount)
+		}
+
+		let nonce = match context {
+			None if invoice_request.metadata().is_some() => None,
+			Some(OffersContext::InvoiceRequest { nonce }) => Some(nonce),
+			_ => return Err(Bolt12ResponseError::VerificationError),
+		};
+
+		let invoice_request = match nonce {
+			Some(nonce) => match invoice_request.verify_using_recipient_data(
+				nonce, expanded_key, secp_ctx,
+			) {
+				Ok(invoice_request) => invoice_request,
+				Err(()) => return Err(Bolt12ResponseError::VerificationError),
+			},
+			None => match invoice_request.verify_using_metadata(expanded_key, secp_ctx) {
+				Ok(invoice_request) => invoice_request,
+				Err(()) => return Err(Bolt12ResponseError::VerificationError),
+			},
+		};
+
+		self.get_response_for_verified_invoice_request(&invoice_request, custom_amount_msats)
+	}
+
+	fn get_response_for_verified_invoice_request(&self, invoice_request: &VerifiedInvoiceRequest, custom_amount_msats: Option<u64>) -> Result<(OffersMessage, PaymentHash), Bolt12ResponseError> {
+		let amount_msats = match InvoiceBuilder::<DerivedSigningPubkey>::amount_msats(
+			&invoice_request.inner, custom_amount_msats
+		) {
+			Ok(amount_msats) => amount_msats,
+			Err(error) => return Err(Bolt12ResponseError::SemanticError(error)),
+		};
```

Comment:
Isn't this also called in the builder?

Reply:
Yes!
But AFAIU, this is also called here so that the `amount_msats` value can be used with the `create_inbound_payment` and `create_blinded_payment_paths` functions.